consultation_analyser/consultations/views/answers.py

```diff
@@ -50,3 +51,98 @@ def index(
     }
 
     return render(request, "consultations/answers/index.html", context)
+
+
+@user_can_see_consultation
+def show(
+    request: HttpRequest,
+    consultation_slug: str,
+    question_slug: str,
+    response_id: int,
+):
+    consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
+    question = get_object_or_404(models.Question, slug=question_slug, consultation=consultation)
+    response = get_object_or_404(models.Answer, id=response_id)
+
+    # TODO: update with QuestionPart
+    all_theme_mappings = models.ThemeMapping.get_latest_theme_mappings_for_question_part(
+        part=response.question_part
+    )
+    all_themes = models.Theme.objects.filter(id__in=all_theme_mappings.values("theme"))
+    existing_themes = models.ThemeMapping.objects.filter(answer=response).values_list(
+        "theme", flat=True
+    )
+
+    if request.method == "POST":
+        requested_themes = request.POST.getlist("theme")
+        existing_mappings = models.ThemeMapping.objects.filter(answer=response)
+
+        # themes to delete
+        existing_mappings.exclude(theme_id__in=requested_themes).delete()
+
+        # themes to add
+        themes_to_add = set(requested_themes).difference([str(theme) for theme in existing_themes])
+        for theme_id in themes_to_add:
+            models.ThemeMapping.objects.create(answer=response, theme_id=theme_id, stance="")
+
+        # flag
+        response.is_theme_mapping_audited = True
+        response.save()
+
+        return redirect(
+            "show_next_response", consultation_slug=consultation_slug, question_slug=question_slug
+        )
+
+    elif request.method == "GET":
+        context = {
+            "consultation_name": consultation.title,
+            "consultation_slug": consultation_slug,
+            "question": question,
+            "response": response,
+            "all_themes": all_themes,
+            "existing_themes": existing_themes,  #  update this key
```

Comment:
Could this comment be removed? (Wasn't sure what it meant.)

Reply:
oops yes. I changed the key and didn't take out the comment